eldritch/test_gates.py

```diff
@@ -647,5 +647,303 @@ def testOther29(self):
     self.assertEqual(self.char.stamina, 4)
 
 
+class Dreamlands22Test(GateEncounterTest):
+  def setUp(self):
+    super().setUp()
+    self.char.place = self.state.places["Dreamlands1"]
+    self.state.event_stack.append(gate_encounters.Dreamlands22(self.char))
+    self.assertEqual(self.char.dollars, 3)
+
+  def testFailWill(self):
+    side_effect = [3, 3] + [1, 1]
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(side_effect=side_effect)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 3)
+
+  def testPassLo(self):
+    side_effect = [3, 5] + [1, 1]
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(side_effect=side_effect)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 5)
+
+  def testPassMid(self):
+    side_effect = [3, 5] + [1, 4]
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(side_effect=side_effect)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 8)
+
+  def testPassHi(self):
+    side_effect = [3, 5] + [6, 6]
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(side_effect=side_effect)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 15)
+
+
+class Dreamlands23Test(GateEncounterTest):
+  def testFail(self):
+    self.char.place = self.state.places["Dreamlands1"]
+    self.state.event_stack.append(gate_encounters.Dreamlands23(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=3)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Dreamlands1")
+
+  def testPass1(self):
+    self.char.place = self.state.places["Dreamlands1"]
+    self.state.event_stack.append(gate_encounters.Dreamlands23(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Dreamlands2")
+
+  def testPass2(self):
+    self.char.place = self.state.places["Dreamlands2"]
+    self.state.event_stack.append(gate_encounters.Dreamlands23(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Lost")
+
```

Comment:
A few more tests I would like to see: Other23, Other24, and Dreamlands28 (make sure being devoured works).